schema/types/volume.go

```diff
@@ -106,12 +114,24 @@ func (v Volume) MarshalJSON() ([]byte, error) {
 }
 
 func (v Volume) String() string {
-	s := fmt.Sprintf("%s,kind=%s,readOnly=%t", v.Name, v.Kind, *v.ReadOnly)
+	s := fmt.Sprintf("%s,kind=%s", v.Name, v.Kind)
 	if v.Source != "" {
-		s = s + fmt.Sprintf(",source=%s", v.Source)
+		s += fmt.Sprintf(",source=%s", v.Source)
 	}
-	if v.Mode != "" && v.UID != -1 && v.GID != -1 {
-		s = s + fmt.Sprintf(",mode=%s,uid=%d,gid=%d", v.Mode, v.UID, v.GID)
+	if v.ReadOnly != nil {
+		s += fmt.Sprintf(",readOnly=%t", *v.ReadOnly)
```

Comment:
Could you just build a slice of strings `strings.Join` them together at the end?